capstone/src/main/java/com/google/sps/servlets/FileUnzipServlet.java

```diff
@@ -112,15 +113,43 @@ public static boolean analyzeApkFeatures(String nameOfApk, Blob blob, String use
 
       while(ze != null) {
         String fileName = ze.getName();
+        long compressedSize = ze.getCompressedSize();
+        long uncompressedSize = ze.getSize();
+
+        // For testing purposes in the console
+        System.out.printf("File %s:\n", fileName);
+        System.out.printf("Entry Compressed Size %d:\n", compressedSize);
+
+        // Handle cases where ZipEntry returns -1 for unknown sizes and 
+        if (uncompressedSize == -1) {
+          compressedSize = 0;
+          uncompressedSize = 0;
+          long startOfFile = is.available();
+          long read = 0;
+          byte[] buffer = new byte[10000];
+
+          // Calculates uncompressed size
+          while ((read = zis.read(buffer, 0, 10000)) > 0) {
+            uncompressedSize += read;
+          }
+
+          // Calculates the compressed size
+          compressedSize = startOfFile - is.available();
+          System.out.printf("Real Compressed Size %d:\n", compressedSize);
```

Comment:
Do we need to keep this here forever? Can / should we switch this to log statements instead of printf? If we turn these into log statements we could monitor them over time!

Reply:
The statements actually serve as log statements that print to the terminal that we can use to monitor over time. We can't really print to the web console from java because java servlets and the web interfaces only communicate through HTTP request and response so that wouldn't work.

Reply:
They serve as log statements but they aren't log statements :) the challenge with doing printf instead of using a logging handler is that:
a) you can't always access or check the console for live servers
b) using typical logging patterns allows us to do things like plug your log output into something like an elastisearch logstash
c) we can control the level of logs - maybe you want to see a bunch of stuff to help you debug, we can set log level to DEBUG, or we can move it to only show you WARNING and up! 
d) logs write out to console if your logging level is set appropriately :D 

Specifically:

```
import java.util.logging.Level; 
import java.util.logging.Logger; 
import java.util.logging.*; 

public class SomeClass {
    private static final Logger LOGGER = Logger.getLogger( ClassName.class.getName() );

    public SomeClass() { 
        LOGGER.info("ZOMG I am constructing!" );
    }

}
```

The switch should be trivialish and it's a good practice. I use printf to debug / play around while I'm coding too but using logging properly is vital for production services. I think these messages could be logger.info level but that's up to you.

Reply:
Okay ma'am. Thank you for the suggestion. I will implement that now.